tests/unit-tests/test-cmd-parser.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+# import utils.sh
+source ../../src/utils.sh
+
+# create flag:variable_name dictionary
+declare -A flag_to_variable_dict
+
+# add flag:variable name dictionary entries
+flag_to_variable_dict[-v]="VERBOSE_LOGGING"
+flag_to_variable_dict[-dp]="DEVICE_PROVISIONING"
+flag_to_variable_dict[-ap]="AZURE_CLOUD_IDENTITY_PROVIDER"
+flag_to_variable_dict[-s]="SCOPE_ID"
+flag_to_variable_dict[-r]="REGISTRATION_ID"
+flag_to_variable_dict[-k]="SYMMETRIC_KEY"
+
+# unit test to test all illegal flags, output should be empty
+test_illegal_flags() {
```

Comment:
>test [](http://example.com/codeflow?start=0&length=4)

do the etsts need to return status code so that the pipeline knows how to flag the run? can you also add them to the pipeline?

Reply:
added to pipeline! status code we can chat about later